sbc_email/models/import_letters_history.py

```diff
@@ -0,0 +1,384 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014-2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emmanuel Mathier, Loic Hausammann <loic_hausammann@hotmail.com>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+"""
+This module reads a zip file containing scans of mail and finds the relation
+between the database and the mail.
+"""
+import logging
+import base64
+import zipfile
+
+from openerp.addons.sbc_compassion.tools import import_letter_functions as func
+from openerp import fields, models, api, _, exceptions
+
+from io import BytesIO
+from openerp.tools.config import config
+from smb.SMBConnection import SMBConnection
+from smb.smb_structs import OperationFailure
+
+from openerp.addons.connector.session import ConnectorSession
+from openerp.addons.sbc_compassion.models import import_letters_history as ilh
+
+logger = logging.getLogger(__name__)
+
+
+class ImportLettersHistory(models.Model):
+    """
+    Keep history of imported letters.
+    This class add to is parent the possibility to select letters to import
+    from a specify config.
+    The code is reading QR codes in order to detect child and partner codes
+    for every letter, using the zxing library for code detection.
+    """
+    _name = 'import.letters.history'
+    _inherit = ['import.letters.history', 'import.letter.config']
+
+    manual_import = fields.Boolean(
+        string="Manual import",
+        default=False)
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+    @api.one
+    @api.onchange("data", "import_folder_path")
+    def _count_nber_letters(self):
+        """
+        Counts the number of scans. If a zip file is given, the number of
+        scans inside is counted.
+        """
+        self.ensure_one()
+
+        if self.manual_import or (
+                self.state and self.state != 'draft'):
+            super(ImportLettersHistory, self)._count_nber_letters()
+        else:
+            # files are not selected by user so we find them on NAS
+            # folder 'Imports' counter
+            tmp = 0
+            # Retrieve configuration
+            smb_user = config.get('smb_user')
+            smb_pass = config.get('smb_pwd')
+            smb_ip = config.get('smb_ip')
+            smb_port = int(config.get('smb_port', 0))
+            if not (smb_user and smb_pass and smb_ip and smb_port):
+                return False
+
+            smb_conn = SMBConnection(smb_user, smb_pass, 'openerp', 'nas')
+            config_obj = self.env['ir.config_parameter']
+            share_nas = (config_obj.search(
```

Comment:
I tried to do it but in method where self.ensure_one() is called, it's doesn't work...
I will try again